### ISSUE TITLE
[go-example] object doesn't exist

### ISSUE THREAD

**Issue by ncarenton** [fsouza/fake-gcs-server#280]
Thanks for this project! 

I can not make the go-example work, it finds the bucket but not the object:

```
docker run -d --name fake-gcs-server -p 4443:4443 -v ${PWD}/examples/data:/data fsouza/fake-gcs-server
763d400668bf735e067fc61b0983a49e4393eb673ae9473379e961faf050a90b
cd examples/go
go build
./go-example 
buckets: [sample-bucket]
2020/06/19 19:31:47 storage: object doesn't exist
```

The strange thing is that curl is working fine so the object is actually there:

```
 curl --insecure https://127.0.0.1:4443/storage/v1/b/sample-bucket/o
{"kind":"storage#objects","items":[{"kind":"storage#object","name":"some_file.txt","id":"sample-bucket/some_file.txt","bucket":"sample-bucket","size":"33","timeCreated":"2020-06-19T17:41:33Z","timeDeleted":"0001-01-01T00:00:00Z","updated":"2020-06-19T17:41:33Z","generation":"0"}]}

```

It is also working fine with the python example:

```
 python python.py 
Bucket: sample-bucket

Blob: some_file.txt
b'Some amazing content to be loaded' 

```

**Comment by fsouza** [comment]
Hey @ncarenton, thanks for reporting and apologies for the delayed response. Not sure if you noticed this comment in the example file: https://github.com/fsouza/fake-gcs-server/blob/c822c3b926f90755b38471be4b77d422d3623007/examples/go/main.go#L5-L9

In order for the downloads to work in the way the Go example expects, the server needs to be started with some flags.

**Comment by ncarenton** [comment]
Hi @fsouza, no worries for the short delay :) 

I actually did miss this comment, it could indeed be moved to `downloadFile()` for example, for better visibility. 

Starting my docker container with `-public-host` does fix my issue! Still I would like to ask if it is possible to use a regular GCS client with the fake server, meaning without these 2 arguments:  `option.WithEndpoint("https://storage.gcs.127.0.0.1.nip.io:4443/storage/v1/"), option.WithHTTPClient(httpClient)` ? Is it possible to do that so we start the fake server without ` -public-host` then ? I get the feeling it would be simpler.

Many thanks for your help

**Comment by fsouza** [comment]
I don't know if the Go sdk supports `STORAGE_EMULATOR_HOST` 🤔 Does it?

If the sdk supports that env var, we could support it in the emulator too, or figure out a way to bind to any host (not sure if it's possible because of conflicting paths).

**Comment by ncarenton** [comment]
I guess so https://github.com/googleapis/google-cloud-go/blob/master/storage/storage.go#L104

**Comment by fsouza** [comment]
Nice, in that case we need to look for that variable and if it's defined, bind in the proper port and use HTTP instead of HTTPs.

**Comment by yiyue115** [comment]
Hello 😃  Actually I faced a similar issue as well cuz I missed that comment 😛 
Is there any update on this one?

**Comment by OrangeFlag** [comment]
Hello, in my case I have set the correct external-url and public-host for my docker container, but when I call fake-gcs from the local machine, I also get a 404

I used wireshark and realized that the requests differ only in the Location field.
Wouldn't it be nice to reply based on the Location being sent and remove the external-url and public-host?
So it would be possible to call the service from any environment

Thanks for your work!
